File: 5 Kyu/Javascript/anagram.js
// function anagram(word, words) {
// 	const wordSorted = word.split('').sort();
// 	return words.filter((e) => e.split('').sort().every((_e, i, a) => a[i] === wordSorted[i]));
// }

function anagram(word, words) {
	const wordSorted = word.split('').sort().join('');
	return words.filter((e) => e.split('').sort().join('') === wordSorted);
}

console.log(anagram('abba', [ 'aabb', 'abcd', 'bbaa', 'dada', 'aaab' ]));

File: 5 Kyu/Python/anagram.py
def anagrams(word, words):
    return list(filter(lambda i: (sorted(i) == sorted(word)), words))

print(anagrams('abba', ['aabb', 'abcd', 'bbaa', 'dada', 'aaab']))
# array = ['aabb', 'abcd', 'bbaa', 'dada']

# print(list(filter(lambda i: i.find('d'), array)))
